pkg/service/project.go

```diff
@@ -79,7 +85,23 @@ func (s *projectService) Update(ctx context.Context, req *v1.ProjectUpdateReques
 func (s *projectService) Delete(ctx context.Context, req *v1.ProjectDeleteRequest) (*v1.ProjectResponse, error) {
 	project := req.NewProject()
 	err := s.projectStore.Delete(ctx, project.Meta.Id)
-	return project.NewProjectResponse(), err
+	if err != nil {
+		return nil, err
+	}
+	filter := map[string]any{
+		"projectmember ->> 'project_id'": project.Meta.Id,
+	}
+	memberships, _, err := s.projectMemberStore.Find(ctx, filter, nil)
```

Comment:
Wouldn't it be better to first delete memberships and then the project to make sure there are no leftovers? As otherwise if a member deletion fails the project is already gone and the call cannot be repeated but maybe there are still project members lying around.

Reply:
You're right, I'll change the order.

Reply:
done